### ISSUE TITLE
psa: nix issues

### ISSUE THREAD

**Issue by benclmnt** [benclmnt/tmc2-rs#3]
if you have problem (related to compiling e.g. some library version, GLIBC version etc) when running `nix shell` try changing `shell.nix` to the following

```nix
with (import <nixpkgs> {
  overlays = [ (import <rust-overlay>) ];
});
stdenv.mkDerivation {
  name = "rust-env";
  nativeBuildInputs = [ rustPlatform.bindgenHook ];
  buildInputs = [
    ffmpeg
    libiconvReal
    pkg-config
    rust-bin.stable.latest.minimal
  ];
```

**Comment by benclmnt** [comment]
- If you're using nix non-root installation and encountered error
	- try using [nix-user-chroot](https://github.com/nix-community/nix-user-chroot)
	- if you get GLIBC v2.33 not found error (`error: /lib/x86_64-linux-gnu/libc.so.6: version 'GLIBC_2.33' not found (required by ...)`)
		- solve this using rust-overlay: https://github.com/oxalica/rust-overlay
	- if you get get libclang.so not found
		- sOlution 2: Add `nativeBuildInputs = [ rustPlatform.bindgenHook ];` to shell.nix [source](https://gist.github.com/yihuang/b874efb97e99d4b6d12bf039f98ae31e)
		- more reading: https://ryantm.github.io/nixpkgs/languages-frameworks/rust/

tldr is given above
